feat: add Chatbase class

tonyghiani opened the pull request:
- get method to retrieve messages
- put method to insert a message

tonyghiani committed:
feat: add Chatbase class
- get method to retrieve messages
- put method to insert a message

nandanrao commented on lib/index.js:
```diff
@@ -0,0 +1,38 @@
+class Chatbase {
+  constructor() {
+    this.pool = new Pool({
+      user: process.env.user || argValidation('user'),
```
Why lowercase env vars?? I would always do uppercase, more canonical. Also, why not use ESNext default arguments if you want what are essentially optional arguments?

tonyghiani replied:
Ops distraction error, my bad! I always follow the upper case convention, i'll switch to default argument!

nandanrao reviewed (commented):
Other than that looks good, let's see some tests!

tonyghiani committed:
wip: add first test and uniqness

tonyghiani committed:
feat: modular util functions

tonyghiani committed:
chore: update dependencies

tonyghiani committed:
cleanup: remove useless commented code

nandanrao commented on lib/index.js:
```diff
@@ -0,0 +1,50 @@
+  async put (key, message) {
+    let exists = { rows: [true]};
+    let hash = '';
+    // Generate a new hash in case an existing one is found as unique id in the database
```
I would push this work to the DB. No point in making multiple requests. Just create the hash and insert it as a column (so 3 columns) -- and set a UNIQUE constraint on the id column (or make it the primary key)

tonyghiani replied:
Yes i was considering to use it as primary key!

nandanrao commented on lib/chatbase.test.js:
```diff
@@ -0,0 +1,41 @@
+const Chatbase = require('./index')
+
+describe('Chatbase Postgres', () => {
+  const mockMessage = {
```
Where did you get this format?

nandanrao replied:
The format is put(key, value) where the value is the JSON data itself. Therefore "message" in your case will be the string data, and this client will be responsible for adding the timestamp.

nandanrao replied:
Maybe for now add timestamp as an optional parameter with default Date.now()  -- 

```
put(key, value, timestamp = Date.now())
```

I'm thinking we should actually store timestamp outself on reception in Kafka...

nandanrao replied:
I take it back. Let's change the format. Make timestamp required: 

```
put(key, value, timestamp)
```

nandanrao committed:
Refactored Chatbase class for v0.0.1

- Use 32bit hash without custom seed
- Tests work with blank Postgres docker image
- Test exceptions
- Test multiple gets
- Enable no args to constructor
- Add timestamp to put command
